code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java

```diff
@@ -149,15 +149,16 @@ void handleRulesEngineResponse(@NonNull final Event event) {
 			return;
 		}
 
-		dispatchTrackRequest(detail, event.getTimestamp());
+		dispatchTrackRequest(detail, event.getTimestamp(), event);
 	}
 
 	/**
 	 * Helper to create and dispatch an experience event.
 	 * @param data map containing free-form data to send to Edge Network
 	 * @param timestamp timestamp of Event
+	 * @param parentEvent the parent event that triggered this request
 	 */
-	private void dispatchTrackRequest(final Map<String, Object> data, final long timestamp) {
+	private void dispatchTrackRequest(final Map<String, Object> data, final long timestamp, final Event parentEvent) {
```

Comment:
we can pass in just the parentEvent and extract the timestamp too: dispatchTrackRequest(final Map<String, Object> data, final Event parentEvent)